### PR TITLE
Fix navigation bug (resolve #224)

### LINKED ISSUE #224: Navbar closes when current link is clicked
Steps to reproduce:
1. navigate to instructor center
2. click on the current link (in the root, it will be `Dashboard`)

Expected outcome:
Nothing to happen. The nav could possibly close and a hamburger show up like in mobile but that's a design choice.

Actual outcome: 
Nav closes with no way, besides a page refresh, to get it back again.

### PR DESCRIPTION
**Pull request by trevordmiller**
Should fix the problem you found @ijones16, thanks for letting me know

### CODE REVIEW

File: src/components/Navigation/index.js [hunk 1]
```diff
@@ -60,152 +54,139 @@ export default class Navigation extends Component {
...
                 color='white'
               />
-            </div>
-
-            <div className='pa2 pa3-ns bb tc b--dark-blue'>
-              <EggoInstructorBanner />
-            </div>
+            )
+
+            if(isFunction(item.action)) {
```
Comment (theianjones) [inline src/components/Navigation/index.js:123]: the logic in this is confusing me. 

`if(isFunction(item.action)` then return:
```
<a
                   key={index}
                   className={sharedLinkClassnames}
                   style={sharedLinkStyle}
                   onClick={() => {
                     if(!isLikelyDesktop) {
                       this.close()
                     }
                     item.action()
                   }}
                 >
                   <NavigationLinkContents />
                 </a>
```
else return a `Navlink` if the `item.action` starts with `/`... but if it doesnt, return an anchor tag (they look like the same `<a/>`?

Comment (theianjones) [reply src/components/Navigation/index.js:123]: Looks like this isn't a part of the changes, just haven't looked at this closely before :D

**Review by theianjones — APPROVED** [review]
Looks good!

File: src/components/Navigation/index.js (hunk 1 above)
Comment (trevordmiller) [reply src/components/Navigation/index.js:123]: @ijones16 Thanks for letting me know; this logic lets you call any "action" from a navigation item click (so run a function, link to another website, or link to a relative link with react router); see here:
https://github.com/eggheadio/egghead-instructor-center/blob/master/src/index.js#L69-L103

image

So, `Log Out` calls a function, `Guide` links to another website (the instructor guide), `Lessons` links to the relative `/lessons` link inside the app.

Make sense?

**Pull request merged**